tests/TestCases/TestCase.php

```diff
@@ -0,0 +1,380 @@
+<?php
+declare(strict_types=1);
+
+namespace Tests\LoyaltyCorp\Multitenancy\TestCases;
+
+use DateInterval;
+use DateTime;
+use Doctrine\Common\Cache\ArrayCache;
+use Doctrine\ORM\EntityManager as DoctrineEntityManager;
+use Doctrine\ORM\EntityManagerInterface as DoctrineEntityManagerInterface;
+use Doctrine\ORM\Tools\SchemaTool;
+use Doctrine\ORM\Tools\Setup;
+use EoneoPay\Externals\Bridge\Laravel\Container;
+use EoneoPay\Externals\Bridge\Laravel\Validator;
+use EoneoPay\Externals\Container\Interfaces\ContainerInterface;
+use EoneoPay\Externals\ORM\EntityManager;
+use EoneoPay\Externals\ORM\Interfaces\EntityManagerInterface;
+use EoneoPay\Externals\Validator\Interfaces\ValidatorInterface;
+use EoneoPay\Utils\Interfaces\Exceptions\ExceptionInterface;
+use EoneoPay\Utils\Interfaces\Exceptions\ValidationExceptionInterface;
+use Exception;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Contracts\Validation\Factory;
+use Laravel\Lumen\Application;
+use Laravel\Lumen\Testing\TestCase as BaseTestCase;
+use PHPUnit\Framework\Constraint\Exception as ExceptionConstraint;
+use PHPUnit\Framework\TestResult;
+use RuntimeException;
+use Tests\LoyaltyCorp\Multitenancy\Helpers\ApplicationBootstrapper;
+use Tests\LoyaltyCorp\Multitenancy\Helpers\DateIntervalFormatter;
+use Throwable;
+
+/**
+ * @noinspection EfferentObjectCouplingInspection
+ *
+ * @coversNothing
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects) Centralised logic for all tests
+ * @SuppressWarnings(PHPMD.ExcessiveClassComplexity) Complexity required for testing
+ * @SuppressWarnings(PHPMD.NumberOfChildren) All tests extend this class
+ * @SuppressWarnings(PHPMD.TooManyFields) Required for base test functionality
+ */
+class TestCase extends BaseTestCase
```

Comment:
Lets not call this TestCase

I'd prefer to see our "TestCase" be just one extending from PHPUnits, and this one can be named AppTestCase